custom_components/unifiprotect/entity.py

```diff
@@ -99,8 +120,70 @@ def _async_updated_event(self) -> None:
 
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass."""
+        await super().async_added_to_hass()
         self.async_on_remove(
             self.protect_data.async_subscribe_device_id(
                 self.device.id, self._async_updated_event
             )
         )
+
+
+class AccessTokenMixin(Entity):
+    """Adds access_token attribute and provides access tokens for use for anonymous views."""
+
+    @property
+    def access_tokens(self) -> collections.deque[str]:
+        """Get valid access_tokens for current entity"""
+        assert isinstance(self, UnifiProtectEntity)
+        return self.protect_data.async_get_or_create_access_tokens(self.entity_id)
+
+    @callback
+    def _async_update_and_write_token(self):
+        _LOGGER.debug("Updating access tokens for %s", self.entity_id)
+        self.async_update_token()
+        self.async_write_ha_state()
+
+    @callback
+    def async_update_token(self) -> None:
+        """Update the used token."""
+        self.access_tokens.append(
+            hashlib.sha256(_RND.getrandbits(256).to_bytes(32, "little")).hexdigest()
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the extra state attributes."""
+        attrs = super().extra_state_attributes or {}
+        return {
+            **attrs,
+            "access_token": self.access_tokens[-1],
+        }
+
```

Comment:
```suggestion
```

Do we need the token separately since its already provided in the URL?

Reply:
That was mirroring how the camera component did it. My assumption for that was it did it to ensure the access token was always in the state the in event of an unexpected shutdown/restart. The last one would not be lost.

Reply:
I'm pretty sure it was added for the frontend to consume it.  I don't think we need it for this case.

Reply:
https://github.com/home-assistant/frontend/blob/dev/src/data/camera.ts#L45